src/v/archival/ntp_archiver_service.cc

```diff
@@ -444,11 +444,12 @@ ntp_archiver::upload_segment(upload_candidate candidate) {
     auto reset_func =
       [this,
        candidate]() -> ss::future<std::unique_ptr<storage::stream_provider>> {
-        co_return std::make_unique<storage::concat_segment_reader_view>(
-          candidate.sources,
-          candidate.file_offset,
-          candidate.final_file_offset,
-          _io_priority);
+        return ss::make_ready_future<std::unique_ptr<storage::stream_provider>>(
```

Comment:
I was looking at this one recently in another PR and thinking of just copying `_io_priority` - then we do not need to capture `this`, so something like

```
    auto reset_func =
      [io_priority = _io_priority,
       candidate]() -> ss::future<std::unique_ptr<storage::stream_provider>> {
        return ss::make_ready_future<std::unique_ptr<storage::stream_provider>>(
          std::make_unique<storage::concat_segment_reader_view>(
            candidate.sources,
            candidate.file_offset,
            candidate.final_file_offset,
            io_priority));
    };
```